sql/src/main/java/io/crate/analyze/AnalyzedTableElements.java

```diff
@@ -256,64 +260,94 @@ void finalizeAndValidate(RelationName relationName,
         validateColumnStorageDefinitions();
     }
 
-    private void validateGeneratedColumns(RelationName relationName,
-                                          Collection<? extends Reference> existingColumns,
-                                          Functions functions,
-                                          ParameterContext parameterContext,
-                                          CoordinatorTxnCtx coordinatorTxnCtx) {
+    private void validateExpressions(RelationName relationName,
+                                     Collection<? extends Reference> existingColumns,
+                                     Functions functions,
+                                     ParameterContext parameterContext,
+                                     CoordinatorTxnCtx coordinatorTxnCtx) {
         List<Reference> tableReferences = new ArrayList<>();
         for (AnalyzedColumnDefinition columnDefinition : columns) {
             buildReference(relationName, columnDefinition, tableReferences);
         }
         tableReferences.addAll(existingColumns);
 
         TableReferenceResolver tableReferenceResolver = new TableReferenceResolver(tableReferences, relationName);
-        ExpressionAnalyzer expressionAnalyzer = new ExpressionAnalyzer(
+        ExpressionAnalyzer generatedExpressionAnalyzer = new ExpressionAnalyzer(
             functions, coordinatorTxnCtx, parameterContext, tableReferenceResolver, null);
+        ExpressionAnalysisContext generatedExpressionAnalysisContext = new ExpressionAnalysisContext();
+
```

Comment:
Maybe clarify why a separate expression analyzer is used.

```suggestion
// Default expressions must not contain column references, so a separate instance with FieldProvider.UNSUPPORTED is used.
```